roles/ams/tasks/deploy.yml

```diff
@@ -46,6 +46,7 @@
     - ams_certs
     - ams_config
   notify: restart argo-messaging
+  when: ams_push_enabled is defined and ams_push_enabled | default(false) | bool
```

Comment:
I guess you can also add the condition to the directory creation aswell, which is the above step
- name: Create CA dir if not exists